### PR TITLE
FRONTEND-0739 :: BUG :: Fix InMemoryDataSource in Harmony Typescript

### PR DESCRIPTION
**Pull request by orioljp**
**Merge / Pull request information:**
Fix and tests for the InMemoryDataSource

* Asana task link:
https://app.asana.com/0/1200574432121296/1202495260510739/f

### CODE REVIEW

**Comment by doup** [comment]
I would wait until #94 is fixed & merged, then continue with this one. Or the other way around, the thing is that the comments regarding test organization from the other PR are also valid for this PR.

**Comment by orioljp** [comment]
@doup these two PRs are not made by the same person, they can be merged indistinctly. Could you check this one? I refactored it following Miguel's instructions

**Review by joselufo — APPROVED** [review]

File: packages/core/tests/repository/data-source/in-memory.data-source.test.ts
Comment (doup) [inline packages/core/tests/repository/data-source/in-memory.data-source.test.ts:55]: I would add line breaks in the tests to denote the given/when/then structure:

```suggestion
            const cacheDataSource = new InMemoryDataSource<SimplestClass>();
            await cacheDataSource.put(myObject, keyQuery);

            await cacheDataSource.delete(keyQuery);

            const result = cacheDataSource.get(keyQuery)
            await expect(result).rejects.toThrow(NotFoundError);
```

**Review by doup — APPROVED** [review]
**Add line breaks to differentiate given/when/then blocks as suggested by Miguel**. Once you have that you can merge. I'm pre-approving so you're not blocked.

**Review by lucianosantana — APPROVED** [review]

File: packages/core/tests/repository/data-source/in-memory.data-source.test.ts
Comment (orioljp) [reply packages/core/tests/repository/data-source/in-memory.data-source.test.ts:55]: Actually, the get is the "when"

**Comment by orioljp** [comment]
@doup I see it totally unnecessary. Let me explain.

- A part from being a matter of code formatting, I added intermediate variables to make it **semantically explicit**.
- **Miguel didn't talk about adding spaces** told _"No need to keep the comments, this is just to make you aware of the 3 sections in the structure."_
- In your example, you are mixing the sections, because the delete is part of the given. We are preparing the scenario there. Is the get which is de "when", that's why I'm always making it **explicit with the variable result**.
- The "then" it's always the expect.

For this reasons, at least in typescript where everything is this clear, I think that it's totally unnecessary adding spaces

**Pull request merged**
